### PR TITLE
LPD-40819 Removed Developer License Price for Cloud Apps

### PR DESCRIPTION
**Pull request by CalebJamesH**
https://liferay.atlassian.net/jira/software/c/projects/LPD/boards/256?quickFilter=4481&selectedIssue=LPD-40819

### CODE REVIEW

**Comment by CalebJamesH** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/ProvideVersionDetailsPage/ProvideVersionDetailsPage.tsx [hunk 1]
```diff
@@ -93,12 +96,24 @@ export function ProvideVersionDetailsPage({
 			({key}) => key === 'dxp-license-usage-type'
 		);
 
-		const targetOption = isDXP ? dxpOption : trialOption;
+		const cloudOption = options.find(
+			({key}) => key === 'cloud-license-usage-type'
+		);
```
Comment (eduardo-diniz) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/ProvideVersionDetailsPage/ProvideVersionDetailsPage.tsx:101]: What do you think to change this `options.find(({key}) => key ===""` into a function?
```
const findOptionByKey = (key: string) => options.find(({ key: k }) => k === key);

const trialOption = findOptionByKey('trial');
const dxpOption = findOptionByKey('dxp-license-usage-type');
const cloudOption = findOptionByKey('cloud-license-usage-type');
```

**Review by eduardo-diniz — CHANGES_REQUESTED** [review]
LGTM :rocket:

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/ProvideVersionDetailsPage/ProvideVersionDetailsPage.tsx [hunk 2]
```diff
@@ -93,12 +96,24 @@ export function ProvideVersionDetailsPage({
...
+
+		const targetOption = isDXP
+			? dxpOption
+			: isCloud
+				? cloudOption
+				: trialOption;
```
Comment (eduardo-diniz) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/ProvideVersionDetailsPage/ProvideVersionDetailsPage.tsx:107]: Please check if  `trialOption` is necessary `const trialOption = options.find(({key}) => key === 'trial');`

**Review by eduardo-diniz — CHANGES_REQUESTED** [review]

**Comment by CalebJamesH** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/ProvideVersionDetailsPage/ProvideVersionDetailsPage.tsx (hunk 2 above)
Comment (CalebJamesH) [reply workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/ProvideVersionDetailsPage/ProvideVersionDetailsPage.tsx]: no problem

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/enums/Product.ts [hunk 3]
```diff
@@ -29,6 +19,11 @@ export enum PRODUCT_IMAGE_FALLBACK_CATEGORIES {
 	PRODUCT_IMAGE = 'productImage',
 }
 
+export enum PRODUCT_LICENSE {
+	DXP = 'dxp-license-usage-type',
+	CLOUD = 'cloud-license-usage-type',
```
Comment (welitonpe) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/enums/Product.ts:24]: Just sort

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/InformLicensingTermsPage/InformLicensingTermsPricePage.tsx [hunk 4]
```diff
@@ -168,37 +171,44 @@ export function InformLicensingTermsPricePage({
...
-							handleAddPriceTier(LicenseTier.DEVELOPER)
-						}
-					>
-						Add Developer Licenses
-					</IconButton>
-				)}
-			</Section>
+			{isDXP && (
+				<Section
+					label="Developer License prices"
```
Comment (welitonpe) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/InformLicensingTermsPage/InformLicensingTermsPricePage.tsx:176]: Could you use translate.
```suggestion
					label={i18n.translate("developer-license-prices")}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/InformLicensingTermsPage/InformLicensingTermsPricePage.tsx [hunk 5]
```diff
@@ -168,37 +171,44 @@ export function InformLicensingTermsPricePage({
...
+					tooltip="Developer licenses are limited to 5 unique addresses and should not be used for full scale production deployments."
```
Comment (welitonpe) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/InformLicensingTermsPage/InformLicensingTermsPricePage.tsx:177]: Same

```suggestion
					tooltip={i18n.translate("developer-licenses-are-limited-to-5-unique-addresses-and-should-not-be-used-for-full-scale-production-deployments")}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/InformLicensingTermsPage/InformLicensingTermsPricePage.tsx [hunk 6]
```diff
@@ -168,37 +171,44 @@ export function InformLicensingTermsPricePage({
...
+							}
+						/>
+					) : (
+						<IconButton
+							className="icon-button py-3 w-100"
+							onClick={() =>
+								handleAddPriceTier(LicenseTier.DEVELOPER)
+							}
+						>
+							Add Developer Licenses
```
Comment (welitonpe) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/InformLicensingTermsPage/InformLicensingTermsPricePage.tsx:207]:
```suggestion
							{i18n.translate(Add Developer Licenses)}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/utils/util.ts [hunk 7]
```diff
@@ -32,6 +32,29 @@ export function createSkuName(
 	}`;
 }
 
+export function getCloudOptionBody() {
+	return {
+		fieldType: 'radio',
+		key: 'cloud-license-usage-type',
```
Comment (welitonpe) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/utils/util.ts:38]: same translate

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/utils/util.ts [hunk 8]
```diff
@@ -32,6 +32,29 @@ export function createSkuName(
...
+	};
+}
+
+export function getCloudProductOptionBody(newOptionId: number) {
+	return {
+		facetable: false,
+		fieldType: 'radio',
+		key: 'cloud-license-usage-type',
+		name: {
+			en_US: 'Cloud License Usage Type',
```
Comment (welitonpe) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/utils/util.ts:49]: same translate

**Review by welitonpe — CHANGES_REQUESTED** [review]

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/enums/Product.ts (hunk 3 above)
Comment (CalebJamesH) [reply workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/enums/Product.ts:24]: thx!

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/InformLicensingTermsPage/InformLicensingTermsPricePage.tsx (hunk 4 above)
Comment (CalebJamesH) [reply workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/InformLicensingTermsPage/InformLicensingTermsPricePage.tsx:176]: thx!

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/InformLicensingTermsPage/InformLicensingTermsPricePage.tsx (hunk 5 above)
Comment (CalebJamesH) [reply workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/InformLicensingTermsPage/InformLicensingTermsPricePage.tsx:177]: thx!

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/InformLicensingTermsPage/InformLicensingTermsPricePage.tsx (hunk 6 above)
Comment (CalebJamesH) [reply workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/PublisherDashboard/pages/Apps/AppCreationFlow/InformLicensingTermsPage/InformLicensingTermsPricePage.tsx:207]: thx!

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/utils/util.ts (hunk 7 above)
Comment (CalebJamesH) [reply workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/utils/util.ts:38]: thx!

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/utils/util.ts (hunk 8 above)
Comment (CalebJamesH) [reply workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/utils/util.ts:49]: thx!

**Comment by CalebJamesH** [comment]
ci:test:sf

**Comment by CalebJamesH** [comment]
ci:test:sf

**Comment by CalebJamesH** [comment]
ci:test:sf

**Comment by CalebJamesH** [comment]
ci:test:sf

**Comment by CalebJamesH** [comment]
ci:test:sf

**Comment by CalebJamesH** [comment]
ci:test:sf

**Review by welitonpe — APPROVED** [review]
LGTM! :fire:

**Review by eduardo-diniz — APPROVED** [review]

**Comment by welitonpe** [comment]
ci:test:stable

**Comment by welitonpe** [comment]
ci:forward

**Pull request closed without merging**